app/src/features/sources/SourceForm.tsx

```diff
@@ -53,9 +51,15 @@ const useStyles = makeStyles((theme) => ({
   },
 }));
 
-const SourceForm = ({ source, submitSuccess }: SourceFormProps) => {
+const SourceForm = () => {
   const { t } = useTranslation();
   const classes = useStyles();
+  const dispatch = useAppDispatch();
+
+  const source = useAppSelector(selectSourceToEdit);
+
+  const openDrawer = !!source;
+  const handleCloseDrawer = () => dispatch(editSource(null));
```

Comment:
Oops.. I guess `null` is used to reset the state.

Do we really need to though ?

Reply:
The drawer opens when we specify a source to edit